### PR TITLE
[EXT-863] Splitting notifications docs, adding some examples

### PR DESCRIPTION
**Pull request by JeremyLawsonReach**
Moving notifications into its own folder, adding some examples. Examples for order declined and failed are still missing, trying to track those down before we add them.

### CODE REVIEW

**Review by sandybrydon — APPROVED** [review]

File: notifications/README.md [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+As part of the on-boarding for the Session API, you will provide Reach a webhook URL. This is where you should
+expect the notifications from Reach to be sent via a Http POST request. After successfully receiving a notification,
+respond to Reach with a HttpStatus of 2XX. If you respond with a HttpStatus not in the 2XX range, Reach will attempt to
+resend the notification. There are six different notifications, two which refer to the Session and four which refer to the
+Order.
+
+Reach does not guarantee the order in which the notifications will arrive. Your server will need to determine the
+relevancy of the notification depending on the last known state on your side.
+
+| Event Type  | How to Handle? |
```
Comment (marcwenger) [inline notifications/README.md:21]: `How to Handle?` to `How to handle transaction in your system` ?

File: notifications/README.md [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+| ------------- | ------------- |
+| `SESSION_FAILED`  | Mark as failed and notify the customer you were unable to process payment. A new Session will need to be created if the user tries again.  |
+| `SESSION_COMPLETED`  | Informational notification letting the merchant know that the customer has completed the “Drop-In” experience. **This notification does not mean the payment has been successful.**  |
+| `ORDER_AUTHORIZED`  | The Order created by the Session was Authorized. If auto-capture is set to false, you can proceed to capture the Order. If auto-capture is set to true, you can ignore this notification.  |
+| `ORDER_PROCESSED`  | Mark as completed/done in your system. The Order has been successfully captured and funds have been moved.  |
```
Comment (marcwenger) [inline notifications/README.md:26]: Remove `in your system`

File: notifications/README.md [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+resend the notification. There are six different notifications, two which refer to the Session and four which refer to the
+Order.
+
+Reach does not guarantee the order in which the notifications will arrive. Your server will need to determine the
+relevancy of the notification depending on the last known state on your side.
+
+| Event Type  | How to Handle? |
+| ------------- | ------------- |
+| `SESSION_FAILED`  | Mark as failed and notify the customer you were unable to process payment. A new Session will need to be created if the user tries again.  |
+| `SESSION_COMPLETED`  | Informational notification letting the merchant know that the customer has completed the “Drop-In” experience. **This notification does not mean the payment has been successful.**  |
```
Comment (marcwenger) [inline notifications/README.md:24]: I like the **bold**, but move `This notification does not...` sentenace to a new line.  Makes it stand out more

File: notifications/README.md (hunk 1 above)
Comment (JeremyLawsonReach) [reply notifications/README.md:21]: Yup that's better.

File: notifications/README.md [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+
+When Reach sends a notification but either no response is received, or the response contains a non-2xx level HTTP
+status, the notification will be queued for a retry after a period of time. Subsequent retries are attempted
+with an exponential backoff with a maximum of 25 attempts - early retries will have a few seconds between each attempt,
+while the last attempt will have approximately 17 hours between attempts (total time in retrying will be within 5 days).
+When a successful HTTP response is received, the retries will halt.
+
+## Missed notifications
+
+If individual notifications are missed, please use the session retrieval endpoint (v1/session/{sessionId}) to retrieve
```
Comment (marcwenger) [inline notifications/README.md:40]: Forward slash `/v1/session...`

File: notifications/README.md [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+| `SESSION_COMPLETED`  | Informational notification letting the merchant know that the customer has completed the “Drop-In” experience. **This notification does not mean the payment has been successful.**  |
+| `ORDER_AUTHORIZED`  | The Order created by the Session was Authorized. If auto-capture is set to false, you can proceed to capture the Order. If auto-capture is set to true, you can ignore this notification.  |
+| `ORDER_PROCESSED`  | Mark as completed/done in your system. The Order has been successfully captured and funds have been moved.  |
+| `ORDER_PROCESSING_FAILED`  | Mark as failed and notify the customer you were unable to process payment. A new Session will need to be created if the user tries again.  |
+| `ORDER_DECLINED`  | Mark as failed and notify the customer you were unable to process payment. A new Session will need to be created if the user tries again.  |
+
+## Notification retries
+
+When Reach sends a notification but either no response is received, or the response contains a non-2xx level HTTP
+status, the notification will be queued for a retry after a period of time. Subsequent retries are attempted
```
Comment (marcwenger) [inline notifications/README.md:33]: Remove `after a period of time`

**Review by tdawatson — APPROVED** [review]

File: notifications/README.md [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+## Missed notifications
+
+If individual notifications are missed, please use the session retrieval endpoint (v1/session/{sessionId}) to retrieve
+the session information as needed. If a large number of notifications are missed, please coordinate with Reach to
+attempt to reschedule sending the missed notifications.
+
+## Notification Verification
+Notifications sent from Reach contain a `reach-signature` header value that can be used to verify that the notification
+is authentic and untampered. The signature is a base-64 encoded HMAC-256 hash using the shared secret key provided to
+the merchant during onboarding. See the [withreach Signature calculation](https://docs.withreach.com/docs/signature-calculation)
```
Comment (marcwenger) [inline notifications/README.md:47]: `withreach` to `With Reach` or `WithReach` or `Reach`

File: notifications/README.md (hunk 5 above)
Comment (JeremyLawsonReach) [reply notifications/README.md:33]: Yeah, I'll rework the next sentence little bit too - I want to make sure it's clear that they're not queued for immediate sending.

**Review by marcwenger — APPROVED** [review]

**Pull request merged**